### PR TITLE
Interactivelist

### PR DESCRIPTION
**Pull request by samuelkinyanjui**
Add a new JavaScript file and import it as a module:
it will contain methods related to the status updates (completed: true / false).
Add event listener to the checkbox (change).
Update items object's value for completed key upon user actions.
Implement a function for the "Clear all completed" button (use filter() method).
Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  {
+    description: 'To do List Structure',
+    completed: true,
+    index: 2,
+  },
+  {
+    description: 'Add and Remove List',
+    completed: false,
+    index: 3,
+  },
```
Comment (tugsem) [inline src/index.js]: - [ ] Please remove all of the hardcoded tasks as it was the previous task's requirement.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    tasks.forEach((task, index) => {
+      addItemsContainer.innerHTML += `
+          <li class="list-item" draggable="true" data-index="${index}">
+              <input type="checkbox" name="${task.description}" ${task.completed ? 'checked' : ''}>
+              <input type="text" value="${task.description}" readonly>
+              <div class="actions-container">
+            <i class="bi bi-three-dots-vertical" data-index="${index}"></i>
+            </div>
+          </li>
+        `;
```
Comment (tugsem) [inline src/index.js]: - [ ] I think there is a problem with your current code. You should render the items from the local storage and should have modules for adding, removing books and updating the local storage. But you only rendering the hardcoded items and adding book is not working.  Please add modules for adding and removing books, then, you should implement these:
- [ ] Add a module for updating status changes
- [ ] Add an event handler for checkboxes
- [ ] Add a module for clearing all completed items.
- [ ] Save all of the changes to the local storage

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @samuelkinyanjui 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
❌ add and remove methods are missing
❌ the project requirements aren't met
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @samuelkinyanjui, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `.github/workflows` for linters ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1,2 +1,155 @@
...
+  cd Capstone-Project_sam
+ npm install
+```
+
+### Usage
+
+To run the project, execute the following command:
+
+> npm install
+
```
Comment (Uzair-Manzoor) [inline README.md:89]: **Incorrect command:** In the Usage section, there is an inconsistency between the provided command and the description. The description mentions running the project with the command `npm install`, which is incorrect. It should be something like `npm run build` and `npm start` instead.

**Comment by samuelkinyanjui** [comment]
Implemented the changes

**Pull request closed without merging**